src/client/telemetry/index.ts

```
@@ -150,6 +150,7 @@ export function captureTelemetry<P extends IEventNamePropertyMapping, E extends
                             properties,
                             ex
                         );
+                        throw ex;
```

Triple check with Don just to make sure swallowing exception was not intentional here. Seems like that would be incorrect to swallow and your change seems right, but this is being used by the main extension as well.

---

I wonder if maybe this works because the result is actually returned as is, and not actual promise from the catch statement?

---
In reply to: [262170466](https://github.com/Microsoft/vscode-python/pull/4632#discussion_r262170466) [](ancestors = 262170466)

---

Meaning I don't actually need this change here because the exception isn't actually swallowed. It's just swallowed on this particular then.catch path.

---
In reply to: [262173031](https://github.com/Microsoft/vscode-python/pull/4632#discussion_r262173031) [](ancestors = 262173031,262170466)